src/command/client/search/interactive.rs

```diff
@@ -229,15 +237,17 @@ impl State {
 
     #[allow(clippy::cast_possible_truncation)]
     fn draw_compact<T: Backend>(&mut self, f: &mut Frame<'_, T>, results: &[History]) {
+        let show_help = f.size().height > 1;
         let chunks = Layout::default()
             .direction(Direction::Vertical)
             .margin(0)
             .horizontal_margin(1)
             .constraints(
                 [
-                    Constraint::Length(1),
+                    Constraint::Length(u16::from(show_help)),
```

Comment:
This was initially as the following (which IMO was more readable because we are talking number of lines):
```suggestion
                    Constraint::Length(if show_help { 1 } else { 0 }),
```

But this happened to trigger the following clippy lint control:
```
warning: boolean to int conversion using if
   --> src/command/client/search/interactive.rs:247:40
    |
247 |                     Constraint::Length(if show_help { 1 } else { 0 }),
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with from: `u16::from(show_help)`
    |
    = note: `show_help as u16` or `show_help.into()` can also be valid options
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_to_int_with_if
    = note: `-D clippy::bool-to-int-with-if` implied by `-D warnings`
```

Just because the help height is precisely 1 line in compact mode. With a height of 2 lines, the clippy lint did not trigger.

Reply:
This no longer is the case on the recent compiler IIRC. anyway, I would prefer the original here, and if the warning comes back I would accept an `allow()`

Reply:
Done (still requires an `allow()`).